sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -966,9 +966,7 @@ object TypeCoercion extends TypeCoercionBase {
       // Implicit cast from/to string
       case (StringType, DecimalType) => DecimalType.SYSTEM_DEFAULT
       case (StringType, target: NumericType) => target
-      case (StringType, DateType) => DateType
-      case (StringType, TimestampType) => TimestampType
-      case (StringType, AnyTimestampType) => AnyTimestampType.defaultConcreteType
```

Comment:
shall we still keep this line? in case a function defines its expected type as `AnyTimestampType`, and we need to cast string to the default concrete type of `AnyTimestampType`

Reply:
updated, thanks